pennylane/measurements/mid_measure.py

```diff
@@ -55,23 +55,72 @@ def func(x, y):
     >>> func(*pars)
     tensor([0.90165331, 0.09834669], requires_grad=True)
 
+    Wires can be reused after measurement. Moreover, measured wires can be reset
+    to the :math:`|0 \rangle` by setting ``reset=True``.
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def func(x, y):
+            qml.RY(x, wires=0)
+            qml.CNOT(wires=[0, 1])
+            m_0 = qml.measure(1, reset=True)
+
+            qml.cond(m_0, qml.RY)(y, wires=0)
+            qml.RX(np.pi/4, wires=1)
+            return qml.probs(wires=[0, 1])
+
+    Executing this QNode:
+
+    >>> pars = np.array([0.643, 0.246], requires_grad=True)
+    >>> func(*pars)
```

Comment:
Might be worth finding an example where we can clearly see the effect of `reset` on the outcome.

Maybe something along the lines of:
```
dev = qml.device("default.qubit", wires=3)
@qml.qnode(dev)
def func():
    qml.PauliX(1)
    m_0 = qml.measure(1, reset=reset)
    return qml.probs(wires=[1])
```